### ISSUE TITLE
Quick Start Provider Example

### ISSUE THREAD

**Issue by nadilas** [zalando/gin-oauth2#46]
Hello,

is there an example provider server, which could be spun up for testing this library with the [zalando](https://github.com/zalando/gin-oauth2/blob/master/example/zalando/main.go) example?

**Comment by szuecs** [comment]
@nadilas in general you can not get any token from Zalando infrastructure. You could use github or google provider if you like to use cloud providers and there is also https://github.com/coreos/dex which is an open source oauth server implementation, which you could use.

**Comment by nadilas** [comment]
@szuecs right, sorry. I meant a basic runnable implementation via dex which works with the examples/zalando code (user, teams, services concept).

**Comment by szuecs** [comment]
@nadilas no sorry, I did not investigated how to do this. It would be nice to have something like this. Also for testing the implementation with unit tests it would be really nice to have something like that. If you want to build it feel free to do so, I am happy to merge or help if there is a problem.
